app/assets/javascripts/modules/testsByCategoryWidget.js

```diff
@@ -0,0 +1,166 @@
+        userIds: $scope.params.userIds
+      };
+
+      if ($scope.params.projectVersion) {
+        params.projectVersionId = $scope.params.projectVersion.id;
+      } else {
+        params.projectIds = $scope.params.projectIds;
+      }
+
+      return api({
+        url: '/metrics/testsByCategories',
+        params: _.extend({}, params, {
+          organizationId: $scope.organization.id
+        })
+      }).then(showMetrics);
+    }
+
+    function showMetrics(response) {
+      // Sum the number of tests for all the categories
+      var total = _.reduce(response.data.categories, function(memo, category) {
+        return memo + category.testsCount;
+      }, response.data.noCategoryTestsCount);
+
+      // Prepare the data for sorting
+      var testsByCategoryRaw = _.clone(response.data.categories);
+
+      // Show the number of tests without a category only if there more than one test
+      if (response.data.noCategoryTestsCount > 0) {
+        testsByCategoryRaw.push({
+          name: 'No category',
+          testsCount: response.data.noCategoryTestsCount
+        });
+      }
+
+      // Sort the metrics by testsCount
+      var testsByCategorySorted = _.sortBy(testsByCategoryRaw, function(testByCategory) {
+        return -testByCategory.testsCount;
+      });
+
+      // Split the categories to show and the categories to group under "other"
+      var slicing = testsByCategorySorted.length > $scope.nbCategories ? $scope.nbCategories : testsByCategorySorted.length - 1;
+      var retainedCategories = testsByCategorySorted.slice(0, slicing);
+      var otherCategories = testsByCategorySorted.slice(slicing);
+
+      var testsByCategory = [];
+
+      // Calculate the percentage of the shown categories
+      _.each(retainedCategories, function(retainedCategory) {
+        testsByCategory.push(_.extend(retainedCategory, {
+          percent: 100 / total * retainedCategory.testsCount
+        }));
+      });
+
+      // If necessary calculate the number of tests for remaining categories and the corresponding percentage
+      if (!_.isEmpty(otherCategories)) {
+        // Create the other category only if there are two or more other categories
+        if (otherCategories.length > 1) {
+          var othersArray = [];
+
+          var others = _.reduce(otherCategories, function (memo, otherCategory) {
+            othersArray.push(_.extend(otherCategory, {
+              percent: 100 / total * otherCategory.testsCount
+            }));
+            return memo + otherCategory.testsCount;
+          }, 0);
+
+          // Add others only if there is at least one test
+          if (others > 0) {
+            testsByCategory.push({
+              name: 'Others',
+              testsCount: others,
+              percent: 100 / total * others,
+              others: othersArray
+            });
+          }
+
+        // As there is only one other category, just show the category in place of others
+        } else {
```

Comment:
Rather than write:

```
if (!_.isEmpty(otherCategories)) {
  if (otherCategories.length > 1) {
  } else {
  }
}
```

Couldn't we write?

```
if (otherCategories.length > 1) {
} else if (otherCategories.length == 1) {
}
```